WultraMobileTokenSDK/Operations/Model/UserOperation/Attributes/WMTOperationAttributeAmount.swift

```diff
@@ -52,23 +52,27 @@ public class WMTOperationAttributeAmount: WMTOperationAttribute {
         case amount, amountFormatted, currency, currencyFormatted, valueFormatted
     }
     
-    public init(label: AttributeLabel, amount: Decimal, currency: String, amountFormatted: String?, currencyFormatted: String?, valueFormatted: String?) {
-        self.amount = amount
-        self.currency = currency
+    public init(label: AttributeLabel, amountFormatted: String, currencyFormatted: String, valueFormatted: String?, amount: Decimal?, currency: String? ) {
         self.amountFormatted = amountFormatted
         self.currencyFormatted = currencyFormatted
         self.valueFormatted = valueFormatted
+        self.amount = amount
+        self.currency = currency
         super.init(type: .amount, label: label)
     }
     
     public required init(from decoder: Decoder) throws {
         
         let c = try decoder.container(keyedBy: Keys.self)
         
-        amount = (try c.decode(Double.self, forKey: .amount) as NSNumber).decimalValue
-        currency = try c.decode(String.self, forKey: .currency)
-        amountFormatted = try? c.decode(String.self, forKey: .amountFormatted)
-        currencyFormatted = try? c.decode(String.self, forKey: .currencyFormatted)
+        // For backward compatibility with legacy implementation, where the `amountFormatted` and `currencyFormatted` values might not be present,
+        // we directly decode from `amount` and `currency`.
+        amountFormatted = try c.decodeIfPresent(String.self, forKey: .amountFormatted) ?? String(c.decode(Double.self, forKey: .amount))
```

Comment:
To deserialize the amount, we should do `(try c.decode(Double.self, forKey: .amount) as NSNumber).decimalValue` or decode as `Decimal` (like in conversion) otherwise we can get floating point precision errors with a lot of decimal places.